documentation/src/main/asciidoc/upgrading/upgrading.asciidoc

```diff
@@ -7,6 +7,16 @@ The Infinispan community
 
 This guide walks you through the process of upgrading Infinispan.
 
+== Upgrading from 9.0 to 9.1
+
+=== (FineGrained)AtomicMap reimplemented
+
+Infinispan now contains a new implementation of both `AtomicMap` and `FineGrainedAtomicMap`, but the semantics has been preserved. The new implementations don't use `DeltaAware` interface but link:../user_guide/user_guide.html#functional_map_api[Functional API] instead.
+
+There are no changes in needed for `AtomicMap`, but it now supports non-transactional use case as well.
+
+`FineGrainedAtomicMap` now uses link:../user_guide/user_guide.html#grouping[Grouping API] and therefore you need to link:../user_guide/user_guide.html#how_do_i_use_the_grouping_api[enable groups in configuration]. Also it holds entries as regular cache entries, plus one cache entry for cached key set (the map itself). Therefore the cache size or iteration/streaming results may differ. Note that fine grained atomic maps are still supported only on transactional caches.
```

Comment:
add 'the' before the grouping api link
'still supported on transactional caches only'